services/src/modules/paramInjection.ts

```diff
@@ -9,27 +10,27 @@ type jwtData = {
     [name: string]: any;
 };
 
-const paramRegex = /{(\w+\.\w+)}/g;
+const paramRegex = /{(source|args|jwt|exports)\.(\w+(\.\w+)*)}/;
 const authzHeaderPrefix = 'Bearer ';
 
 function resolveTemplate(
-    template: string,
+    source: string,
+    key: string,
     parent: any,
     args: GraphQLArguments,
     context: RequestContext,
     info: GraphQLResolveInfo
 ) {
-    const [sourceName, propName] = template.split('.');
-
-    switch (sourceName.toLowerCase()) {
+    const propPath = key.split('.');
+    switch (source) {
         case 'source':
-            return parent && parent[propName];
+            return parent && R.path(propPath, parent);
         case 'args':
-            return args && args[propName];
+            return args && R.path(propPath, args);
         case 'exports':
-            return context.exports.resolve(info.parentType, parent, propName);
+            return context.exports.resolve(info.parentType, parent, key);
         case 'jwt':
-            return getJwt(context)[propName];
+            return getJwt(context)[key];
```

Comment:
```suggestion
            return R.path(propPath, getJwt(context));
```

We can support the deep nesting here as well.
`getJwt` will return an empty object if there is not JWT so there is no need for an initial existence check.